### ISSUE TITLE
Howto exit full screen article mode and related questions and suggestions

### ISSUE THREAD

**Issue by xandro0777** [yanus171/Handy-News-Reader#354]
In a very sleepy mode I managed to enable it by accident and confirm the dialog without reading the instructions how to switch back to normal mode. I can switch back to article overview and feed overview views but can't find the option to disable the full screen article mode - although there is something with a highly suggestive name (settings/article's text/auto-fullscreen entries) it is unchecked and toggling it back and forth has seemingly no effect in this situation.

How do I do that?

Is there some quick way to access the article menu from (eg share, mark unread..) from full screen mode? Could it be displayed on double tap or long tap?

As a suggestion - when pulling down to make the back button appear again would it be possible to make a full screen toggle available? Or add a button at the end of text next to "reload" ?

**Comment by yanus171** [comment]
There is a standard Android OS gesture to temporary edit full screen mode: swipe from the top of the screen to bottom

**Comment by xandro0777** [comment]
Yes.. luckily remembered the swipe down which saved my life. However, after the swipe down I was able to exit app or return to the list of articles and feeds but whenever I went to another article I was in full screen mode again. This would not be a big problem except that the article menu was unavailable and some of its functionality seems only available from there.

Longer story..  figured out what went wrong: previously I have disabled "Tap Zones" by setting their size to 0. Never used them and as more often than not I tapped one of them by accident they were more of an annoyance than a feature. Except.. by disabling Tap Zones I apparently lost any way to exit the article full screen mode. Apparently it is possible to enter article full screen mode through the article menu but not to exit it anymore unless Tap Zones are enabled?

The solution in my case was the swipe down, followed by twice "back" to return to the feed list, config/operation - reenable Tap Zones here, go to some article and tap the zone to toggle the full screen view here.

I think either the "full screen" option should not be offered in the article menu if Tap Zones are disabled or some other way to toggle full screen should be offered in such cases.

Also there is a clear bug in my version: when full screen has been enabled either in article menu or in settings/article's text/auto-fullscreen , disabling it by unsetting settings/article's text/auto-fullscreen does not work. Checkbox unchecked but articles are still fullscreened.

**Comment by yanus171** [comment]
May be we should better find a solution to show menu in full screen mode?

**Comment by xandro0777** [comment]
Long tap seems unused in article mode and many browsers use it to pop up some menu so this would seem intuitive?

**Comment by yanus171** [comment]
Long tap is used by select text context menu. I could add an item to it to show menu, but would it be intuitive?

**Comment by xandro0777** [comment]
Ok, didn't realize it is by long tap,  some apps use long tap + move/drag for that and in practice those are sometimes hard to distinguish with my fingers.

Accessing the menu from the text selection bar would certainly work for me .. in fact I already looked for it there.

Double tap would be another option, perhaps less intuitive but I guess it is also one of the things which many people  would quickly try in that situation.

**Comment by xandro0777** [comment]
Another possibility would be to recognize when users disabled tap zones (or even  provide a specific option for disabling them) and use single tap - either to switch back from full screen mode or to popup the menu.

For my special case that would be of course the best solution.

**Comment by yanus171** [comment]
Recently I have realized that the app lack a possibility to open menu in full screen mode. The modern phones lack the menu button. And I have a rooted phone with custom room and use one of the hardware buttons as a menu one. 
So in this version  https://drive.google.com/file/d/1Chz1YaBEflv2aeV-s08t87ZS0i7DSEIr/view?usp=drivesdk if action bar is hidden, the menu opens by the tap on empty (non link) space in article text.

**Comment by yanus171** [comment]
And I also added the items in the article menu to toggle full screen mode and action bar visibility

**Comment by xandro0777** [comment]
Thanks, getting the menu or out of the article works fine.

It would be nice if the device "back" key would work directly to leave the arrticle view, not the popup menu. Or alternatively.. it remains visible long enough after leaving the menu.
